src/main/java/org/elasticsearch/index/mapper/core/AbstractFieldMapper.java

```diff
@@ -360,6 +368,14 @@ protected String defaultPostingFormat() {
     protected String defaultDocValuesFormat() {
         return null;
     }
+    
+    protected boolean defaultDocValues() {
```

Comment:
Should we make this method either abstract or return false to avoid traps? For instance I was worried that CompletionFieldMapper does not override it to return false. Maybe it works since completion fields are tokenized so the default value is false but I think making it explicit would be less error-prone?